app/updatemanager.go

```diff
@@ -125,9 +138,137 @@ func (u *UpdateManager) run(ctx context.Context) error {
 	return nil
 }
 
+type UpdateControlMapState struct {
+	Action           string `json:"action"`
```

Comment:
One idea could be to have these as `types`, and then implement custom JSON unmarshaling on them, but how you've done it is just as fine :)

Reply:
We are relying on default unmarshaling for most other JSON objects like `AuthReqData`, `LogData`, etc, so I'll keep it simple here too.